src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -0,0 +1,490 @@
+        // rows/cols, and remove each one from the old grid, and attach it to
+        // our grid instead.
+        while (remainingChild->_root.ColumnDefinitions().Size() > 0)
+        {
+            auto col = remainingChild->_root.ColumnDefinitions().GetAt(0);
+            remainingChild->_root.ColumnDefinitions().RemoveAt(0);
+            _root.ColumnDefinitions().Append(col);
+        }
+        while (remainingChild->_root.RowDefinitions().Size() > 0)
+        {
+            auto row = remainingChild->_root.RowDefinitions().GetAt(0);
+            remainingChild->_root.RowDefinitions().RemoveAt(0);
+            _root.RowDefinitions().Append(row);
+        }
+
+        // Remove the child's UI elements from the child's grid, so we can
+        // attach them to us instead.
+        remainingChild->_root.Children().Clear();
+
+        _root.Children().Append(_firstChild->GetRootElement());
+        _root.Children().Append(_separatorRoot);
+        _root.Children().Append(_secondChild->GetRootElement());
+
+        // Set up new close handlers on the children, so thay'll notify us
+        // instead of their old parent.
+        _SetupChildCloseHandlers();
+
+        // If the closed child was focused, transfer the focus to it's first sibling.
+        if (closedChild->_lastFocused)
+        {
+            _FocusFirstChild();
+        }
+
+        // Release the pointers that the child was holding.
+        remainingChild->_firstChild = nullptr;
+        remainingChild->_secondChild = nullptr;
+        remainingChild->_separatorRoot = { nullptr };
+    }
+}
+
+// Method Description:
+// - Adds event handlers to our chilcren to handle their close events.
+void Pane::_SetupChildCloseHandlers()
+{
+    _firstClosedToken = _firstChild->Closed([this](){
+        _root.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [=](){
+            _CloseChild(true);
+        });
+    });
+
+    _secondClosedToken = _secondChild->Closed([this](){
+        _root.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [=](){
+            _CloseChild(false);
+        });
+    });
+}
+
+// Method Description:
+// - Vertically split the focused pane in our tree of panes, and place the given
+//   TermControl into the newly created pane. If we're the focused pane, then
+//   we'll create two new children, and place them side-by-side in our Grid.
+// Arguments:
+// - profile: The profile GUID to associate with the newly created pane.
+// - control: A TermControl to use in the new pane.
+void Pane::SplitVertical(const GUID& profile, const TermControl& control)
+{
+    // If we're not the leaf, recurse into our children to split them.
+    if (!_IsLeaf())
+    {
+        if (_firstChild->_HasFocusedChild())
+        {
+            _firstChild->SplitVertical(profile, control);
+        }
+        else if (_secondChild->_HasFocusedChild())
+        {
+            _secondChild->SplitVertical(profile, control);
+        }
+
+        return;

```

Comment:
So if neither has a focused child, you just bail? Is this an error scenario that should be logged?